src/components/Table/RowState.ts

```diff
@@ -180,6 +180,16 @@ export class RowState {
       } else {
         collapsedIds.splice(i, 1);
       }
+
+      // If all rows have been expanded individually, but the 'header' was collapsed, then remove the header from the collapsedIds so it reverts to the expanded state
+      if (collapsedIds.length === 1 && collapsedIds[0] === "header") {
+        collapsedIds.splice(0, 1);
+      } else {
+        // If every top level child has been collapsed, then splice the header to the front of the array to be considered collapsed as well.
```

Comment:
Nit that comment says splice, probably written before you fixed the "header must be first" vs. ".includes("header")" line.

Reply:
Ah yes... I had spliced it in at first and kept the check looking for 'header' as the first in the array, but afterwards felt like checking if it was anywhere in the array was a bit safer. Will update the comment.